pkg/testutil/require.go

```diff
@@ -293,11 +296,90 @@ func RequireDecommissionNode(t *testing.T, sb testenv.DiffingSandbox, b ClusterB
 			t.Log("statefulset replicas do not match")
 			return false, nil
 		}
+		err = makeDrainStatusChecker(t, sb, b, uint64(numNodes))
+		if err != nil {
+			t.Logf("makeDrainStatusChecker failed due to error %v\n", err)
+			logPods(context.TODO(), sts, cluster, sb, t)
+			return false, err
+		}
+		return true, nil
+	})
+	require.NoError(t, err)
+}
+
+// RequireDecommisionDrainStatusNode checks that proper nodes are decommisioned using node status cmd
+func RequireDecommisionDrainStatusNode(t *testing.T, sb testenv.DiffingSandbox, b ClusterBuilder, numNodes int32) {
+	err := wait.Poll(10*time.Second, 400*time.Second, func() (bool, error) {
+		err := makeDrainStatusChecker(t, sb, b, uint64(numNodes))
+		if err != nil {
+			t.Logf("makeDrainStatusChecker failed due to error %v\n", err)
+			return false, err
+		}
 		return true, nil
 	})
 	require.NoError(t, err)
 }
 
+func makeDrainStatusChecker(t *testing.T, sb testenv.DiffingSandbox, b ClusterBuilder, numNodes uint64) error {
+	cluster := b.Cluster()
+	cmd := []string{"/cockroach/cockroach", "node", "status", "--decommission", "--format=csv", cluster.SecureMode(), fmt.Sprintf("--port=%d", *cluster.Spec().GRPCPort)}
+
+	stdout, stderror, err := kube.ExecInPod(sb.Mgr.GetScheme(), sb.Mgr.GetConfig(), cluster.Namespace(),
+		fmt.Sprintf("%s-0", cluster.StatefulSetName()), resource.DbContainerName, cmd)
+	if err != nil {
+		t.Logf("exec cmd = %s on pod= %s-0 exit with error %v and stdError %s", cmd, cluster.StatefulSetName(), err, stderror)
+		return err
+	}
+	if stderror != "" {
+		t.Logf("exec cmd = %s on pod= %s-0 exit with error %v and stdError %s", cmd, cluster.StatefulSetName(), err, stderror)
+		return err
+	}
+	r := csv.NewReader(strings.NewReader(stdout))
+	// skip header
+	if _, err := r.Read(); err != nil {
+		return err
+	}
+	for {
+		record, err := r.Read()
+		if err == io.EOF {
+			break
+		}
+
+		if err != nil {
+			return errors.Wrapf(err, "failed to get node draining status")
+		}
+
+		idStr, isLive, replicasStr, isDecommissioning := record[0], record[8], record[9], record[10]
+		id, err := strconv.ParseUint(idStr, 10, 32)
+		if err != nil {
+			return errors.Wrap(err, "failed to extract node id from string")
+		}
+		if id <= numNodes {
```

Comment:
So the logic is here that we always decommission the node with the highest ID first? This is a bit of magic to me, but maybe its fine. For a later PR but maybe we add a check in here to make sure no other nodes are decommissioning?